api/specs.proto

```diff
@@ -101,6 +102,7 @@ message TaskSpec {
 	oneof runtime {
 		NetworkAttachmentSpec attachment = 8;
 		ContainerSpec container = 1;
+		CustomSpec custom = 10;
```

Comment:
@aluzzardi I think I understand the encoding issue. The bigger issue here is that the thing is called "custom", but will likely be used most commonly. _Custom_ usually denotes something that is exceptional or bespoke, where as we are trying to communicate that this merely a generic type holder.